softioc/softioc.py

```diff
@@ -352,3 +352,14 @@ def interactive_ioc(context = {}, call_exit = True):
 
     if call_exit:
         safeEpicsExit(0)
+
+def non_interactive_ioc():
+    '''Function to run the IOC in non-interactive mode. This mode is useful for
+    running the IOC as a background process without user interaction.
+    This function expects a stop signal. When it receives one, the IOC stops.
+    '''
+    if device.dispatcher:
+        device.dispatcher.wait_for_quit()
+        safeEpicsExit(0)
+    else:
+        print("No dispatcher found")
```

Comment:
We don't think this check is necessary, so please remove the if statement entirely.